platform/darwin/src/MGLStyle.h

```diff
@@ -610,6 +610,16 @@ MGL_EXPORT
  */
 @property (nonatomic, strong) MGLLight *light;
 
+#pragma mark Managing Style's localization
+
+/**
+ Set to true to determine the localization according to the languages
+ supported by <a href="https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview">Mapbox Streets</a>, 
+ it will look for the Preferred Language Order setting in the Language & Region section of Settings. 
+ If it doesn't fing a match it will fall back to english localization.
```

Comment:
> What happens if the style does not have the Mapbox Streets source?

None of the style layers would pass [this test](https://github.com/mapbox/mapbox-gl-native/blob/eeff3ccb8c634b66c77a0f7aedc8a0ce2fc855e9/platform/darwin/src/MGLStyle.mm#L628), so there would be no effect on the style.